include/hydrogen/device/gpu/cuda/CUDALaunchKernel.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef HYDROGEN_DEVICE_GPU_CUDALAUNCHKERNEL_HPP_
+#define HYDROGEN_DEVICE_GPU_CUDALAUNCHKERNEL_HPP_
+
+#include <cuda_runtime.h>
+
+#include <hydrogen/Device.hpp>
+#include <hydrogen/SyncInfo.hpp>
+
+#include "CUDAError.hpp"
+
+namespace hydrogen
+{
+namespace gpu
+{
+
+template <typename F, typename... Args>
+void LaunchKernel(
+    F kernel, dim3 const& gridDim, dim3 const& blkDim,
+    size_t sharedMem, SyncInfo<Device::GPU> const& si,
+    Args... kernel_args)
+{
+    void* args[] = { reinterpret_cast<void*>(&kernel_args)... };
```

Comment:
In case the user passes in a `const` arg:
```suggestion
    void* args[] = { const_cast<void*>(reinterpret_cast<const void*>(&kernel_args))... };
```